migrations.go

```diff
@@ -12,6 +12,21 @@ func runDatabaseMigrations(db *gorm.DB) error {
 		&Branch{}, &Build{}, &Log{},
 		&Artifact{}, &BuildParam{}, &Param{}}
 
+	// since v4.2.1, drop these constraints to refresh the constraint behavior.
+	// Previously it was RESTRICT, now it's CASCADE.
+	//
+	// Perform before AutoMigrate because we want them readded in that step.
+	if err := dropOldConstraints(db, []constraintToDrop{
+		{"artifact", "fk_artifact_build"},
+		{"log", "fk_log_build"},
+		{"build", "fk_build_project"},
+		{"log", "fk_log_build"},
+		{"branch", "fk_project_branches"},
+		{"build_param", "fk_build_params"},
+	}); err != nil {
+		return err
+	}
+
```

Comment:
So for each migration, which is run every time the wharf-api restarts, the above constraints are removed for a short period.

I assume you added this because GORM doesn't alter the existing constraints, yes?

Highly suggest to instead to take a more surgical approach. Maybe time to start using yet another table for manual migration steps. A table with either a list of migrations applied, or just a single row with an integer that specifies a "migration revision" that the migrations look for. Thoughts?

Reply:
Or if you could do it in a single transaction for deleting those constraints and then doing the "AutoMigrate" when db.DisableForeignKeyConstraintWhenMigrating is disabled

Reply:
Transitioning to a migrations table would be a good idea for another PR I think. That would be a lot more flexible and maintainable.

I will change it to do it in a single transaction 👍

Reply:
Done in 8a3a54d